include/souper/Infer/Interpreter.h

```diff
@@ -48,14 +51,20 @@ struct EvalValue {
     return Value;
   }
 
-  static EvalValue poison() {
+  static EvalValue poison(unsigned Width) {
+    if (Width == 0)
```

Comment:
since there is not supposed to be a situation where this happens, please make it an assertion, not a regular error check (error checks are for things that can happen)

Reply:
this is to make sure that even if we do a release build without assertions, it doesn't go unnoticed. I vaguely remember some time back when we changed the asserts to this "fatal error" to make sure we don't skip asserts under any circumstances.

Reply:
souper has a lot of assertions in it, we never got rid of them

Reply:
if you're simply guarding against internal inconsistency, assertion is usualyl the way to go, and it is what you should be using here

Reply:
ok. i will change it to assert.